# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1387867324 18000
#      Tue Dec 24 01:42:04 2013 -0500
# Node ID 8d332da6694fb32645018e21f22b2deb58c9fe00
# Parent  441121728230e373c30034d59e89edccf75702b9
# Parent  002c06b3fc51e41d8805b81682639643245125e1
maint: Periodic merge of gui-release to default.

diff --git a/run-octave.in b/run-octave.in
--- a/run-octave.in
+++ b/run-octave.in
@@ -73,17 +73,23 @@ if [ $# -gt 0 ]; then
     driver="strace -o octave.trace"
     shift
   elif [ "x$1" = "x-cli" ]; then
     octave_executable="$builddir/src/octave-cli"
     shift
   fi
 fi
 
+## We set OCTAVE_ARCHLIBDIR so that the wrapper program can find the
+## octave-gui program in the build tree.  That will fail if we ever
+## need Octave to find other things in ARCHLIBDIR that are not built
+## in the $builddir/src directory.
+
 OCTAVE_BINDIR="$builddir/src" \
+OCTAVE_ARCHLIBDIR="$builddir/src" \
 OCTAVE_SITE_INITFILE="$top_srcdir/scripts/startup/main-rcfile" \
 OCTAVE_DEFAULT_QT_SETTINGS="$builddir/libgui/default-qt-settings" \
 OCTAVE_LOCALE_DIR="$builddir/libgui/languages" \
 OCTAVE_JAVA_DIR="$builddir/scripts/java" \
   exec $builddir/libtool --mode=execute $driver \
     "$octave_executable" --no-init-path --path="$LOADPATH" \
     --image-path="$IMAGEPATH" --doc-cache-file="$DOCFILE" \
     --built-in-docstrings-file="$BUILT_IN_DOCSTRINGS_FILE" \
